shell/platform/android/io/flutter/embedding/engine/dart/DartExecutor.java

```diff
@@ -196,6 +196,26 @@ public void send(@NonNull String channel, @Nullable ByteBuffer message, @Nullabl
   public void setMessageHandler(@NonNull String channel, @Nullable BinaryMessenger.BinaryMessageHandler handler) {
     messenger.setMessageHandler(channel, handler);
   }
+
+  /**
+   * Returns the number of pending channel callback replies.
+   *
+   * <p>When sending messages to the Flutter application using {@link BinaryMessenger#send(String,
+   * ByteBuffer, io.flutter.plugin.common.BinaryMessenger.BinaryReply)}, devlopers can optional
+   * specify a reply callback if they expect a reply from the Flutter application.
+   *
+   * <p>This method tracks all the pending callbacks that are waiting for response, and is supposed
+   * to be called from the main thread (as other methods). Calling from a different thread could
+   * possibly capture an indeterministic internal state, and is strongly discouraged.
+   *
+   * <p>Currently, it's mainly useful for a testing framework like Espresso to determine whether all
+   * the async channel callbacks are properly handled and the app is idle.
```

Comment:
Out of curiosity, does this ensure they are "properly handled"? or just that it's idle?

Reply:
I removed "properly". I feel it's too vague to simply say "it's idle", because usually it doesn't equal completely idle. Rather, it's a state we define as idle.